### PR TITLE
Fixed reloading of the toolbar modules.

### PR DESCRIPTION
**Pull request by tipue-dev**
## Proposed change

If the reload for pages is activated by an agent, a page- and a toolbar-reload will be active.
image
With this feature enabled, we get the following problems:

- The toolbar is needlessly reloaded on pages that are fully reloaded in the same time span.
- The current code identifies the toolbar modules by their classes, which doesn't work for CustomerUser search and the Customer search, and results in them being replaced by full-text search fields instead.
- The search profile drop-down isn't modernized after the refresh

I also think to create a full header and cutting it down with RegEx isn't a very elegant solution.

https://user-images.githubusercontent.com/80031303/201231334-2a82c4af-ec1c-484e-86ae-a0f91e1ccd02.mp4

1 - 🐞 bug 🐞

## Checklist

- [x] The code change is tested and works locally.(❗)
- [x] There is no commented out code in this PR.(❕)
- [ ] You improved or added new unit tests.(❕)
- [x] Local ZnunyCodePolicy passed.(❕)
- [x] Local UnitTests / Selenium passed.(❕)
- [x] GitHub workflow CI (UnitTests / Selenium) passed.(❗)

### CODE REVIEW

**Comment by tipue-dev** [comment]
The tests that failed for two of my commits are different and one of those ran without problems on my fork on GitHub and locally, so i **assume** it is not because of my code, but the tests just don't run perfectly yet.

File: Kernel/Modules/AgentDashboardCommon.pm [hunk 1]
```diff
@@ -486,30 +486,19 @@ sub Run {
...
-        TOOLBARITEM:
-        for my $ToolbarItem (@ToolbarItems) {
-            next TOOLBARITEM if $ToolbarItem eq 'UserAvatar';
-
-            ( my $ToolbarItemCSS ) = $LayoutObject->Header() =~ m{<li class="$ToolbarItem">(.*?)<\/li>}s;
-
-            $ToolbarItems{Icons}->{$ToolbarItem} = $ToolbarItemCSS;
+        my $ToolBarModule = $ConfigObject->Get('Frontend::ToolBarModule');
+        my $ToolbarItems;
+        if ( ref $ToolBarModule eq 'HASH' ) {
```
Comment (dennykorsukewitz) [inline Kernel/Modules/AgentDashboardCommon.pm]:
```suggestion
        if ( IsHashRefWithData($ToolBarModule) ) {
```

File: Kernel/Output/HTML/Layout.pm [hunk 2]
```diff
@@ -1590,6 +1502,111 @@ sub Header {
     return $Output;
 }
 
+sub ToolbarModules {
+    my ( $Self, %Param ) = @_;
+
+    my $MainObject = $Kernel::OM->Get('Kernel::System::Main');
```
Comment (dennykorsukewitz) [inline Kernel/Output/HTML/Layout.pm]:
```suggestion
    my $MainObject   = $Kernel::OM->Get('Kernel::System::Main');
```

File: Kernel/Output/HTML/Layout.pm [hunk 3]
```diff
@@ -1590,6 +1502,111 @@ sub Header {
...
+    my $ConfigObject = $Kernel::OM->Get('Kernel::Config');
+
+    my %Modules;
+    my %Jobs = %{$Param{ToolBarModule}};
```
Comment (dennykorsukewitz) [inline Kernel/Output/HTML/Layout.pm]:
```suggestion
    my %Jobs = %{ $Param{ToolBarModule} };
```

File: Kernel/Output/HTML/Templates/Standard/HeaderToolbar.tt [hunk 4]
```diff
@@ -0,0 +1,52 @@
+# --
+# Copyright (C) 2001-2021 OTRS AG, https://otrs.com/
+# Copyright (C) 2021-2022 Znuny GmbH, https://znuny.org/
```
Comment (dennykorsukewitz) [inline Kernel/Output/HTML/Templates/Standard/HeaderToolbar.tt]:
```suggestion
# Copyright (C) 2021 Znuny GmbH, https://znuny.org/
```

**Review by dennykorsukewitz — CHANGES_REQUESTED** [review]
Hi  @tipue-dev ,
thanks for your PR.

I have a few change requests here.
But all in all, wonderful again. :+1: :rocket:

This fixes the errors that occurred: https://github.com/znuny/Znuny/commit/3eec62be923b35af8a89e86a5bf3060846010f2d

Regards :rocket:

**Review by dennykorsukewitz — APPROVED** [review]

**Pull request merged**